src/main/java/com/google/sps/servlets/MemesServlet.java

```diff
@@ -81,11 +81,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
     Bucket bucket = storage.get(BUCKET_NAME);
     Page<Blob> blobs = bucket.list();
-
+    String imgTag = "";
     // Output <img> elements as HTML.
     for (Blob blob : blobs.iterateAll()) {
-      String imgTag = String.format("<img src=\"%s\" width = \"300\"/>", blob.getMediaLink());
-      response.getWriter().println(imgTag);
+      imgTag = imgTag + String.format("<a href=\"" + "https://spring21-sps-43.appspot.com/share?meme=" + blob.getName() + "\"><img src=\"%s\" width = \"300\"/></a>", blob.getMediaLink());
     }
+    response.getWriter().println(imgTag);
```

Comment:
I like the idea of storing the imgTag as a string elsewhere, it helps clean up the code here on line 89